ddtrace/tracer/tracer.go

```diff
@@ -79,9 +79,15 @@ type tracer struct {
 	// pid of the process
 	pid int
 
-	// These integers track metrics about spans and traces as they are started,
-	// finished, and dropped
-	spansStarted, spansFinished, tracesDropped uint32
+	// These maps keep count of the number of spans started and finished from
+	// each component, including contribs and "manual" spans.
+	spansStarted, spansFinished struct {
+		mu    sync.Mutex
+		spans map[string]uint32
+	}
```

Comment:
I do wonder whether:
1. You should use sync.Map instead
2. Because spans are started and finished so often, is the requirement to lock going to cause more contention (and thus degrade performance) than this integration tag is worth?

Maybe @darccio can weigh in on this.